src/login/logind.c

```diff
@@ -328,11 +328,18 @@ static int manager_enumerate_users(Manager *m) {
 
         FOREACH_DIRENT(de, d, return -errno) {
                 User *u;
+                uid_t uid;
 
                 if (!dirent_is_file(de))
                         continue;
 
-                k = manager_add_user_by_name(m, de->d_name, &u);
+                k = safe_atou(de->d_name, &uid);
+                if (k < 0) {
+                        r = log_warning_errno(k, "Failed to add user by file name %s, ignoring: %m", de->d_name);
```

Comment:
maybe use a more specific error message, i.e. `"Failed to parse filename /run/systemd/users/%s as UID.", de->d_name);`

or so.

Reply:
Updated.